gap/Forms.gi

```diff
@@ -0,0 +1,90 @@
+# Note that <group> must have the attribute InvariantBilinearForm or
+# InvariantSesquilinearForm for this to work.
+InstallGlobalFunction("ChangeFixedSesquilinearForm",
+function(group, type, gramMatrix)
+    local gapForm, newForm, gapToCanonical, canonicalToNew, field;
+    if not type in ["S", "O", "U"] then
+        ErrorNoReturn("<type> must be one of 'S', 'U', 'O', but <type> = ",
+                      type);
+    fi;
+    field := DefaultFieldOfMatrixGroup(group);
+    if type = "S" or type = "O" then
+        gapForm := BilinearFormByMatrix(InvariantBilinearForm(group).matrix, 
+                                        field);
+        newForm := BilinearFormByMatrix(gramMatrix, field);
+    else
+        gapForm := HermitianFormByMatrix(InvariantSesquilinearForm(group).matrix,
+                                         field);
+        newForm := HermitianFormByMatrix(gramMatrix, field);
+    fi;
+    # the following if condition can only ever be fulfilled if <group> is an
+    # orthogonal group; there the case of even dimension is problematic since,
+    # in that case, there are two similarity classes of bilinear forms
+    if not WittIndex(gapForm) = WittIndex(newForm) then
+       ErrorNoReturn("The form preserved by <group> must be similar to the form ", 
+                     "described by the Gram matrix <gramMatrix>.");
+    fi;
+    gapToCanonical := BaseChangeHomomorphism(BaseChangeToCanonical(gapForm), 
+                                             field);
+    canonicalToNew := BaseChangeHomomorphism(BaseChangeToCanonical(newForm) ^ (-1), 
+                                             field);
+    return Group(canonicalToNew(gapToCanonical(GeneratorsOfGroup(group))));
+end);
+
+ConjugateModule := function(M, q)
+  return GModuleByMats(List(MTX.Generators(M), A -> ApplyFunctionToEntries(A, x -> x ^ q)), 
+                       MTX.Field(M));
+end;
+
+# Assuming that the group G acts absolutely irreducibly, try to find a unitary
+# form which is G-invariant or prove that no such form exists.
+InstallGlobalFunction("UnitaryForm",
+function(G)
+    local d, F, q, M, hermitianConjugateM, formMatrix, row, col, x, scalar;
+
+    d := DimensionOfMatrixGroup(G);
+    F := DefaultFieldOfMatrixGroup(G);
+    if not IsSquareInt(Size(F)) or not IsPrimePowerInt(Size(F)) then
```

Comment:
The size of any finite field is a prime power, so this could be changed to
```suggestion
    if not IsSquareInt(Size(F)) then
```
However, it'd be slightly more efficient to just check that the degree over the prime field is even:
```suggestion
    if not IsFinite(F) or not IsEvenInt(DegreeOverPrimeField(F)) then
```

Reply:
Done!